docs/reference/specifications/proposal/rfc-grpc-custom-name-resolver.md

```diff
@@ -0,0 +1,157 @@
+# gRPC Custom Name Resolver Proposal (DRAFT)
+
+## Details
+
+|                        |                                    |
+|------------------------|------------------------------------|
+| **Feature Name**       | gRPC custom name resolver          |
+| **Type**               | enhancement                        |
+| **Related components** | gRPC source resolution             |
+
+## Summary
+
+gRPC by default support dns resolution which is currently being used e.g. "localhost:8013" in both
+[core](https://github.com/open-feature/flagd/blob/main/core/pkg/sync/grpc/grpc_sync.go#L72-L74) and
+providers e.g. [java](https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/ChannelBuilder.java#L53-L55).
+This covers most deployments but with increase adoption of microservice architecture and service discovery
+application with S2S policy e.g. istio, envoy, consul etc. In such deployments the current gRPC connection was
+breaking because the proxies need the destination domain i.e. service name in order to route the traffic.
+
+For such cases the gRPC core libs support few alternative resolver* also expose required interface to build custom one
+you can find more details in the document referenced below
+
+### Reference
+
+* [Custom Name Resolution](https://grpc.io/docs/guides/custom-name-resolution/)
+* [Java Client](https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#forTarget(java.lang.String))
+* [Golang](https://pkg.go.dev/google.golang.org/grpc#NewClient)
+
+A sample workflow of a deployment with proxy sidecar using `in-process` mode (for `rpc` the process is same)
+
+```mermaid
+sequenceDiagram
+    participant flagd-sync.service
+    participant proxy-sidecar-agent
+    participant flagd-provider
+    participant application
+
+
+    flagd-provider-->proxy-sidecar-agent: Route traffic to backend sync service based on policy
+    Note left of flagd-provider: In most cases the connection made to <br> `localhost` data plane port <br> e.g. localhost:9211
+
+    flagd-provider-->flagd-sync.service: Get realtime flag update over gRPC stream
+    Note left of proxy-sidecar-agent: Based on the host header i.e. authority <br> the tarrfic was then routed <br> backend service.
+    loop
+        flagd-provider->>flagd-provider: in-memory cache
+    end
+    application-->>flagd-provider: Check the state of a feature flag
+    flagd-provider-->>application: Get the feature flag from in-memory cache <br/> run the evaluation logic and return final state
+```
```

Comment:
This diagram is a bit unclear to me due to the lack of directional arrows.

Something more like https://grpc.io/docs/guides/custom-name-resolution/#life-of-a-target-string makes sense to me.

Reply:
do you mean replace the diagram with the above one ?

Reply:
This is good!